app/components/results/results_component.html.erb

```diff
@@ -0,0 +1,36 @@
+<%= render LocationSubjectFilterComponent.new(results: results) %>
+
+<h1 class="govuk-heading-xl" data-qa="heading">
+  <%= "#{results.number_of_courses_string} found" %>
+</h1>
+
+<div class="app-filter-toggle"></div>
+
+<div class="app-filter-layout">
+  <div class="app-filter-layout__filter">
+    <%= render Results::FilterComponent.new(results: results) %>
+  </div>
+  <div class="app-filter-layout__content">
+    <% if results.no_results_found? %>
```

Comment:
Should the responsibility for deciding whether to render or not be moved to the component's `#render` method?